### DISCUSSION TITLE
need help with timelapse

### DISCUSSION THREAD

**Discussion by alenko** [gopro/labs#981] [General]
I want to do 3 pictures per ady 9:00 13:00 17:00 everyday for infinite...when i want to end it i will just grab camera and end it...

Is there possible to o script. turn on camera at 8:50 take picture at 9:00 and thn shutdown the camera?

**Comment by dnewman-gpsw** [comment]
Yes, easy, but this is not a bug or issue, so I've moved it to discussions. All Labs camera did this has one of the first features.

```dP!09:00S!1N!13:00S!1N!17:00S!1N!R```

This is the whole script.

dP - default Photo - optional if you already default the camera booting to a photo mode.
!09:00S - sleep until at 9am and then press the shutter.
!1N - pause one second (helps with reliability, but will not be necessary in future firmware)
!13:00S - sleep until at 1pm and then press the shutter.
!17:00S - sleep until at 5pm and then press the shutter.
!R - repeat the whole script.


I had the same type of script running for three months (only changed the internal battery twice.)

This is what I'm using (requires GPS or manually setting your location)

```dP!s-1800S!1N!r1800S!1N!12:00S!1N!R```

!s-1800S - sleep until 30 minutes before sunset and then press the shutter.
!r1800S - sleep until 30 minutes after sunrise and then press the shutter.
!12:00S - sleep until noon and then press the shutter.

https://github.com/gopro/labs/assets/10856084/15f47bec-0e5c-4c6e-9293-5afedceadaad

**Comment by alenko** [comment]
Yes, thank you very much, but this doesnt work like if you want to shutdown camera and bringit back up. So it shits down in between. Or if battery is empty and you jist charge it, script will stop right?

**Reply by dnewman-gpsw** [reply]
Sorry I don't understand this: "doesnt work like if you want to shutdown camera and bringit back up".  You can script most actions, I'm just setting the timer for when to wake.  I think is what you asked for.

**Reply by alenko** [reply]
Sory commenting on phone, fat fingers.

if camera shutsdown, does the script turn it on, or camera has to be always on?

**Reply by dnewman-gpsw** [reply]
If your script shuts-down the camera, the camera is completely off, but will wake automatically as requested by the script.  This alarm timer does require the internal battery or external power, but the camera is off, only using enough power that preserves the date and time (the camera's clock.) If you are running a script, and you need to change the battery, you will have to manually power on the camera once, and the script will continue from that point automatically (setting the alarm again as needed.) Or you can add the $WAKE=2 extension, and the battery change will boot the camera, and any previous script will run automatically. However, if you shutdown with the power button, the Labs script is not running, so no alarm was set, it will not wake.
